src/metrics/coranking/script.R

```diff
@@ -30,8 +30,11 @@ if (any(is.na(X_emb))) {
   # TODO: computing a square distance matrix is problematic for large datasets!
   # TODO: should we use a different distance metric for the high_dim?
   # TODO: or should we subset to the HVG?
-  dist_highdim <- coRanking:::euclidean(as.matrix(high_dim))
-  dist_emb <- coRanking:::euclidean(as.matrix(X_emb))
+  message("Compute high-dimensional distances")
+  dist_highdim <- proxyC::dist(high_dim, method = "euclidean")
+  diag(dist_highdim) <- 0 # Force diagonal to be 0
+  message("Compute embedding distances")
+  dist_emb <- parallelDist::parallelDist(as.matrix(X_emb))
```

Comment:
It seems proxyC is faster on dense matrices after all when setting `diag = TRUE`:

![Screenshot from 2024-10-21 16-48-58](https://github.com/user-attachments/assets/1bf294b8-6722-4abc-9ad7-44ea843649cd)